db/row_cache.hh

```diff
@@ -370,8 +371,9 @@ public:
                                      tracing::trace_state_ptr trace_state = nullptr,
                                      streamed_mutation::forwarding fwd = streamed_mutation::forwarding::no,
                                      mutation_reader::forwarding fwd_mr = mutation_reader::forwarding::no,
-                                     const tombstone_gc_state* gc_state = nullptr) {
-        if (auto reader_opt = make_reader_opt(s, permit, range, slice, gc_state, std::move(trace_state), fwd, fwd_mr)) {
+                                     const tombstone_gc_state* gc_state = nullptr,
+                                     max_purgeable_fn get_max_purgeable = {}) {
```

Comment:
Let's avoid more default arguments. Better to explicitly say we don't want gc. I see we have can_always_purge and can_never_purge which convey the intent better.

Reply:
Done in v3.